modules/transactionpool/transactionpool.go

```diff
@@ -97,6 +99,12 @@ func New(cs modules.ConsensusSet, g modules.Gateway, persistDir string) (*Transa
 		persistDir: persistDir,
 	}
 
+	if build.DEBUG {
+		sub := newSanityCheckSubscriber()
+		tp.subscribers = append(tp.subscribers, sub)
+		tp.sanityCheck = sub
```

Comment:
Have `newSanityCheckSubscriber` take a tpool as an argument and then have it call `Subscribe` on the tpool, instead of doing it manually after initialization.